packages/framework/src/Model/Transport/Transport.php

```diff
@@ -244,6 +254,44 @@ public function setPosition($position)
         $this->position = $position;
     }
 
+    /**
+     * @return \Shopsys\FrameworkBundle\Model\Transport\TransportDomain[]
+     */
+    public function getDomains()
+    {
+        return $this->domains;
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Transport\TransportData $transportData
+     */
+    private function setDomains(TransportData $transportData)
+    {
+        foreach ($transportData->domains as $domainId) {
+            try {
+                $paymentDomain = $this->getTransportDomain($domainId);
+            } catch (TransportDomainNotFoundException $e) {
+                $paymentDomain = new TransportDomain($this, $domainId);
+            }
+            $this->domains[$domainId] = $paymentDomain;
```

Comment:
TransportDomain is not an entity for multidomain attributes (similar to ProductDomain), but a optional association with a domain (similar to PaymentDomain). This means, that you should be able to remove the association to a domain (when trying to switch the transport from the first domain to the second, for example), but the setDomains method doesn't allow it, it only adds new Transport domains.